### PR TITLE
test(engine): verify start event with forms

### LINKED ISSUE #14343: Add tests to verify start event behaviour for Forms
**Description**

Add those tests:

* Verify that a process can be deployed with a start event that has a `formId` (assert start event contains formId as an extension)
* Verify that a process can be deployed with a start event that has a `formId` even if form with that id is not deployed
* Verify that a process instance that contains start event with `formId` can be started and completed successfully

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

This PR add tests to verify the correct behaviour of start event element with a form

## Related issues

closes #14343 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/StartEventFormTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  private static final String PROCESS_ID = "process";
+  private static final String TEST_FORM = "/form/test-form-1.form";
+  private static final String FORM_ID = "Form_0w7r08e";
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @Test
+  public void shouldDeployProcessWithFormId() {
```
Comment (berkaycanbc) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/StartEventFormTest.java:41]: 💭  Would be more explanatory if we rename this one to something like `shouldDeployProcessWithFormIdWhenFormIsNotYetDeployed`.

**Review by berkaycanbc — APPROVED** [review]
Very clear, good job @nicpuppa! 🚀 

I only left a minor comment. Leaving the change up to you.

File: engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/StartEventFormTest.java (hunk 1 above)
Comment (nicpuppa) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/StartEventFormTest.java]: Changed 👍

**Comment by nicpuppa** [comment]
bors merge

**Pull request merged**